lib/segment/src/index/hnsw_index/graph_layers_builder.rs

```diff
@@ -77,7 +82,12 @@ impl GraphLayersBuilder {
             .map(|l| l.into_iter().map(|l| l.into_inner()).collect())
             .collect();
 
-        let links = TGraphLinks::from_vec(unlocker_links_layers, path)?;
+        let mut links_converter = GraphLinksConverter::new(unlocker_links_layers);
+        if let Some(path) = path {
```

Comment:
Ideally, this one also should not have a `save` side-effect, but I though it would be too much changes